capsules/extra/src/kv_driver.rs

```diff
@@ -102,7 +102,9 @@ impl<'a, K: kv_system::KVSystem<'a, K = T>, T: kv_system::KeyType> KVSystemDrive
                                                 let static_buffer_len =
                                                     buf.len().min(unhashed_key.len());
 
-                                                // Copy the data into the static buffer
+                                                // Copy the data into the
+                                                // cleared static buffer.
+                                                buf.fill(0);
```

Comment:
I assume we expect these to be pretty small such that filling in slots that we're about to write over in the next line is the same or faster (rather than _only_ clearing the left over after the copy) because of something-something vectorization.

Reply:
Yeah I have them set at 64 bytes.

Really I think the fix is to add a length parameter so we aren't using the entire buffer unconditionally. https://github.com/tock/tock/issues/3495

Reply:
fine with me